### ISSUE TITLE
[a11y]: Missing heading markup in Course Overview accordions impacts screen reader usability

### ISSUE THREAD

**Issue by peterkulko** [openedx/frontend-app-learning#1876]
## Description

The show/hide accordions on the Course Overview screen lack markup which can significantly enhance the user experience for students who use a screen reader.  Specifically, accordions are not marked up as headings to enable quick navigation. Adding heading markup to the accordion labels enables a person who uses a screen reader to jump to a specific accordion using heading navigation.

PR: https://github.com/openedx/frontend-app-learning/pull/1811
